packages/iov-core/src/servercore.spec.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Address,
+  Algorithm,
+  Amount,
+  ChainId,
+  PublicIdentity,
+  PublicKeyBytes,
+  SendTransaction,
+  TokenTicker,
+} from "@iov/bcp-types";
+import { bnsCodec, bnsConnector } from "@iov/bns";
+import { Ed25519, Random } from "@iov/crypto";
+import { Ed25519HdWallet, HdPaths, LocalIdentity, UserProfile } from "@iov/keycontrol";
+
+import { MultiChainSigner } from "./multichainsigner";
+import { ServerCore } from "./servercore";
+
+function pendingWithoutBnsd(): void {
+  if (!process.env.BNSD_ENABLED) {
+    pending("Set BNSD_ENABLED to enable bnsd-based tests");
+  }
+}
+
+async function randomBnsAddress(): Promise<Address> {
+  const rawKeypair = await Ed25519.makeKeypair(await Random.getBytes(32));
+  const randomIdentity: PublicIdentity = {
+    chainId: "some-testnet" as ChainId,
+    pubkey: {
+      algo: Algorithm.Ed25519,
+      data: rawKeypair.pubkey as PublicKeyBytes,
+    },
+  };
+  return bnsCodec.identityToAddress(randomIdentity);
+}
+
+function copyToPublic(local: LocalIdentity): PublicIdentity {
+  return {
+    chainId: local.chainId,
+    pubkey: local.pubkey,
+  };
+}
+
+describe("ServerCore", () => {
+  const bnsdUrl = "ws://localhost:22345";
+
+  const defaultAmount: Amount = {
+    quantity: "1",
+    fractionalDigits: 9,
+    tokenTicker: "CASH" as TokenTicker,
+  };
+
+  it("can be constructed", () => {
+    const profile = new UserProfile();
+    const signer = new MultiChainSigner(profile);
+    const core = new ServerCore(profile, signer);
```

Comment:
This seems a bit of a stutter when constructing it.

What about `const core = new ServerCore(profile)` and have it instantiate the MultiChainSigner inside of it? Or are there other use cases for the internal signer?

Reply:
You need all the signer methods available to be able to connect to chains and from that you need chain IDs to create identities:

```ts
  const profile = new UserProfile();
  const ed25519Wallet = profile.addWallet(Ed25519HdWallet.fromMnemonic(bnsdFaucetMnemonic));
  const secp256k1Wallet = profile.addWallet(Secp256k1HdWallet.fromMnemonic(ganacheMnemonic));
  const signer = new MultiChainSigner(profile);

  const bnsdConnection = (await signer.addChain(bnsConnector(bnsdUrl))).connection;
  const bnsdChainId = bnsdConnection.chainId();
  const ethereumConnection = (await signer.addChain(ethereumConnector(ethereumUrl, undefined))).connection;
  const ethereumChainId = ethereumConnection.chainId();

  // faucet identity
  await profile.createIdentity(ed25519Wallet.id, bnsdChainId, HdPaths.simpleAddress(0));
  // ganache second identity
  await profile.createIdentity(secp256k1Wallet.id, ethereumChainId, HdPaths.bip44(60, 0, 0, 1));

  const serverCore = new ServerCore(profile, signer);
```

I don't see a great way to simplify that right now but maybe ideas will come when working with that code a little more